Fix #3971: [RTL] Add style for all the TextViews.

veena14cs opened the pull request:
## Explanation
This PR fixes #3971 . This PR  ensures that all the Textviews have style attribute set, to generally manage the text alignment for RTL/LTR where the texts are aligned to  right or left and the texts that are center-aligned.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

veena14cs committed:
Update selection_interaction_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Update selection_interaction_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Update return_to_topic_button_item.xml

veena14cs committed:
Update return_to_topic_button_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Update image_region_selection_interaction_item.xml

veena14cs committed:
Update image_region_selection_interaction_item.xml

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
fixed style

veena14cs committed:
Update profile_reset_pin_activity.xml

veena14cs committed:
Merge branch 'develop' into section-topic-style

rt4914 commented on app/src/main/res/layout-land/topic_practice_header_view.xml:
```diff
@@ -19,24 +19,23 @@
       android:id="@+id/master_skills_text_view"
+      style="@style/Heading2"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-land/topic_practice_header_view.xml:
```diff
@@ -19,24 +19,23 @@
       android:id="@+id/skills_description_text_view"
+      style="@style/Subtitle1"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-sw600dp-land/topic_lessons_title.xml:
```diff
@@ -17,12 +17,10 @@
       android:id="@+id/topic_play_text_view"
+      style="@style/Heading2"
       android:layout_width="528dp"
       android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml:
```diff
@@ -22,24 +22,23 @@
         android:id="@+id/master_skills_text_view"
+        style="@style/Heading2"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml:
```diff
@@ -22,24 +22,23 @@
         android:id="@+id/skills_description_text_view"
+        style="@style/Subtitle1"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml:
```diff
@@ -17,12 +17,10 @@
       android:id="@+id/topic_play_text_view"
+      style="@style/Heading2"
       android:layout_width="528dp"
       android:layout_height="wrap_content"
```
layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout-sw600dp-port/topic_practice_header_view.xml:
```diff
@@ -22,24 +22,23 @@
         android:id="@+id/master_skills_text_view"
+        style="@style/Heading2"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`

veena14cs replied:
Done.

rt4914 commented on app/src/main/res/layout/profile_reset_pin_activity.xml:
```diff
@@ -62,14 +62,13 @@
             android:id="@+id/profile_reset_pin_main"
+            style="@style/Body"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
```
layout_width and layout_height can be removed from this and other files as it is `wrap_content`


Ditto for all other files.

veena14cs replied:
Done.

BenHenning replied:
It seems this was wrong per the latest changes (& was reverted). @rt4914 can you confirm whether this comment is still relevant per the latest changes?

rt4914 replied:
This comment can be closed now. Thanks.

rt4914 reviewed (changes requested):
@veena14cs PTAL

veena14cs committed:
removed width and height

rt4914 reviewed (approved):
LGTM, thanks.

BenHenning commented on app/src/main/res/layout-land/topic_practice_header_view.xml:
```diff
@@ -19,24 +19,19 @@
       android:text="@string/topic_practice_skills_description"
       android:textColor="@color/oppiaSecondaryText"
-      android:textSize="16sp"
+      android:textAlignment="viewStart"
```
One of the goals with having central styles is to ensure that all RTL properties are defined through the styles rather than on the text view. Is this something that can be correctly moved to Subtitle1 (per other uses of it) or should this text view use a different style?

veena14cs replied:
To support LTR/RTL and center alignment I have changed the same versions of style with textalignment and gravity attribute. Please check if this makes sense. Once these file changes are approved I will make changes to other files in other PRs.
Note: All the styles for example `Heading1` will be changed to `HeadingCenter1` and `HeadingViewStart1` so final will have only 2 versions of Heading1.

Please let me know if we can propose this solution regardless of other attributes the focus is on only center alignment and right/left alignments .

BenHenning replied:
Per the latest changes, the solution seems good.

BenHenning reviewed (changes requested):
Thanks @veena14cs. This & other PRs actually go into a lot more detail than I was expecting--all I think we actually need to do is ensure RTL-specific properties are defined in styles. The other properties can be moved later once we have a chance to figure out exactly what styles should exist at the top level.

FWIW I'm fine with keeping the current code (other than my comments about text alignment & other RTL properties), but I wanted to clarify expectations in case they weren't clear.

Same feedback for your other PRs. Please assign all back once addressed.

veena14cs committed:
add style to support center align and start align

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into section-topic-style

BenHenning commented on app/src/main/res/layout-land/topic_lessons_title.xml:
```diff
@@ -16,14 +16,10 @@
       android:id="@+id/topic_play_text_view"
-      android:layout_width="wrap_content"
-      android:layout_height="wrap_content"
+      style="@style/Heading2"
```
Doesn't this need to be a view start or center variant? Ditto elsewhere.

(In general, I'd expect all text views to have to pick some view start or view center variant to be explicit about their style).

veena14cs replied:
Done.

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -202,10 +202,25 @@
   </style>
 
-  <style name="Heading3" parent="TextView.Common1">
+  <style name="HeadingViewStart2" parent="TextView.Common1">
```
Suggest inheriting from Heading2 and just defining the new property. Ditto for the other new properties.

veena14cs replied:
Done,

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -202,10 +202,25 @@
+    <item name="android:textAlignment">viewStart</item>
+  </style>
+
+  <style name="HeadingCenter3" parent="TextView.Common1">
```
Suggest HeadingViewCenter3 to have parity with the view start variant. Ditto for the others.

veena14cs replied:
Done.

BenHenning reviewed (changes requested):
Thanks @veena14cs. Left some high-level thoughts, but I think this is definitely on the right track.

My other high-level comment is that we should also be adding generic styles like TextViewCenter and TextViewStart that just define the center/view start properties (to be used by TextViews which don't have styles set today to ensure that all TextViews are covered by centrally-managed RTL styles). While this will add a lot more boilerplate styles, we can resolve these as we clean up styles as part of other ongoing projects.

@rt4914 please let me know if you have any concerns with this.

veena14cs committed:
fixed styles

veena14cs committed:
add missing styles

veena14cs committed:
add mossing styles

veena14cs commented:
> Thanks @veena14cs. Left some high-level thoughts, but I think this is definitely on the right track.
> 
> My other high-level comment is that we should also be adding generic styles like TextViewCenter and TextViewStart that just define the center/view start properties (to be used by TextViews which don't have styles set today to ensure that all TextViews are covered by centrally-managed RTL styles). While this will add a lot more boilerplate styles, we can resolve these as we clean up styles as part of other ongoing projects.
> 
> @rt4914 please let me know if you have any concerns with this.

I have added missing styles and fixed styles that have alignment properties. PTAL.

rt4914 reviewed (commented):
@BenHenning Let me know your thoughts on this.

`Heading2ViewStart` should be named as `Heading2.ViewStart`
`Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`

Same for all other styles.

BenHenning commented on app/src/main/res/layout-land/ongoing_topic_item.xml:
```diff
@@ -57,18 +57,17 @@
           android:layout_marginEnd="8dp"
           android:ellipsize="end"
           android:fontFamily="sans-serif"
-          android:gravity="start"
```
TextViewStart only specifies textAlignment; should the gravity property stay or does it also need to be part of the style? Though, I am wondering whether this is even used given that the text view is in a ConstraintLayout.

veena14cs replied:
It is not being used so removed it.

BenHenning commented on app/src/main/res/layout/profile_chooser_profile_view.xml:
```diff
@@ -79,6 +80,7 @@
 
         <TextView
           android:id="@+id/profile_is_admin_text"
+          style="@style/TextViewCenterHorizontal"
```
Isn't this conditional? Ditto for profile_last_visited.

veena14cs replied:
How to set conditional style for the views? I found that is is not doable https://code.google.com/archive/p/android-developer-preview/issues/2613.

veena14cs replied:
I created new style called `SubtitleLight` that matches the Textviews property without having gravity attribute.

BenHenning replied:
The style thing is a bit of a conundrum. I think defaulting the property is probably the best that we can do, but it's probably going to end up causing some confusion when we figure out the long-term styles & add the CI check. Can you add a comment here & to other versions of this text view (ie in other versions of this file) to explain why this style is used vs. one that defines the text alignment?

veena14cs replied:
Added comment please check if that sounds good.

BenHenning replied:
It looks good--thanks!

BenHenning commented on app/src/main/res/layout-sw600dp-port/profile_chooser_fragment.xml:
```diff
@@ -43,8 +43,7 @@
         <TextView
-          style="@style/Subtitle1"
-          android:gravity="center"
+          style="@style/Subtitle1ViewCenterVertical"
```
Given that android:gravity="center" is both horizontal & vertical, do we need a style that sets both for these cases?

veena14cs replied:
Yes added `Subtitle1Center` style.

BenHenning commented on app/src/main/res/layout-sw600dp-land/ongoing_topic_item.xml:
```diff
@@ -55,17 +55,16 @@
           android:layout_marginEnd="16dp"
           android:ellipsize="end"
           android:fontFamily="sans-serif"
-          android:gravity="start"
```
Both gravity & text alignment is used here--which is correct?

veena14cs replied:
android:gravity="start" doesn't work for RTL. This line doesn't seem to make any difference it is unused.

BenHenning commented on app/src/main/res/layout-land/section_title.xml:
```diff
@@ -25,14 +25,10 @@
 
     <TextView
       android:id="@+id/section_title_text_view"
-      android:layout_width="wrap_content"
-      android:layout_height="wrap_content"
```
Why are these removed? They aren't explicitly defined in the style.

Ditto elsewhere.

veena14cs replied:
Reverted.

BenHenning commented on app/src/main/res/layout-land/section_title.xml:
```diff
@@ -25,14 +25,10 @@
       android:layout_marginEnd="72dp"
-      android:fontFamily="sans-serif-medium"
-      android:text="@{viewModel.sectionTitleText}"
-      android:textColor="@color/oppiaPrimaryText"
```
This also doesn't seem to be defined in the style.

veena14cs replied:
Reverted

BenHenning commented on app/src/main/res/layout/profile_reset_pin_activity.xml:
```diff
@@ -62,14 +62,11 @@
 
           <TextView
             android:id="@+id/profile_reset_pin_main"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
```
Body uses different layout width/height properties than this--is it the correct style to be using? This TextView also didn't define an explicit font family before.

veena14cs replied:
Fixed style for this view.

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -339,4 +413,21 @@
+    <item name="android:textAlignment">viewStart</item>
+  </style>
+
+  <style name="TextViewCenter" parent="TextView.Common1">
```
Would 'CenterVerticalHorizontal' be clearer to distinguish from the other two center options? Ditto for other similar styles.

veena14cs replied:
Should I rename it to `TextViewCenterVerticalHorizontal` or just `CenterVerticalHorizontal`?

veena14cs replied:
Also, I think `TextViewCenter` seems like more general as it defines by default both the vertical and horizontal axis.

BenHenning replied:
Hmm after taking another pass, I think this is fine for the medium term.

BenHenning reviewed (changes requested):
Thanks @veena14cs! This looks much closer to what I was thinking. Just had a few follow-up comments--PTAL.

BenHenning commented:
> @BenHenning Let me know your thoughts on this.
> 
> `Heading2ViewStart` should be named as `Heading2.ViewStart` `Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`
> 
> Same for all other styles.

@rt4914 sure; I don't have a strong stance here since the styles are temporary (though given that they may be around for a bit while we figure out the final way to define styles, it seems reasonable).

@veena14cs please also update the PR to address the latest conflicts.

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into section-topic-style

veena14cs committed:
Update section_title.xml

veena14cs committed:
add style with center gravity

veena14cs commented:
> > @BenHenning Let me know your thoughts on this.
> > `Heading2ViewStart` should be named as `Heading2.ViewStart` `Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`
> > Same for all other styles.
> 
> @rt4914 sure; I don't have a strong stance here since the styles are temporary (though given that they may be around for a bit while we figure out the final way to define styles, it seems reasonable).
> 
> @veena14cs please also update the PR to address the latest conflicts.

@rt4914 I agree with @BenHenning as this is just a temporary solution and I would keep it as it is for now.

rt4914 reviewed (approved):
Sounds good. Lets keep it as it is for now. Thanks.

veena14cs commented:
> Sounds good. Lets keep it as it is for now. Thanks.

Thanks @rt4914 .

veena14cs committed:
added style

veena14cs committed:
added missing style

BenHenning commented on app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml:
```diff
@@ -55,6 +55,7 @@
 
         <TextView
           android:id="@+id/add_profile_text"
+          style="@style/TextViewCenter"
```
This isn't being centered in the layout--is this correct?

veena14cs replied:
No, fixed it.

BenHenning commented on app/src/main/res/layout/profile_progress_header.xml:
```diff
@@ -90,6 +90,7 @@
 
         <TextView
           android:id="@+id/ongoing_topics_count"
+          style="@style/TextViewCenter"
```
Ditto here--this didn't have a center property before.

veena14cs replied:
To keep it consistent I added center property here. And by default it was center aligned based on the layout. Same with other files.

BenHenning replied:
Ah I see--that wasn't very clear. Thanks for clarifying.

BenHenning commented on app/src/main/res/layout-land/profile_chooser_profile_view.xml:
```diff
@@ -61,26 +61,20 @@
 
         <TextView
           android:id="@+id/profile_last_visited"
+          style="@style/SubtitleLight1"
```
Shouldn't this be a 'Center'? Ditto for other profile_chooser_profile_view layouts, and for profile_is_admin_text below.

veena14cs replied:
No, as the gravity is handled conditionally.

BenHenning commented on app/src/main/res/layout/profile_chooser_fragment.xml:
```diff
@@ -100,7 +99,7 @@
-        style="@style/Subtitle1"
+        style="@style/Subtitle1ViewCenter"
         android:layout_marginBottom="24dp"
         android:gravity="center"
```
Can this be removed per the latest style?

veena14cs replied:
Done.

BenHenning reviewed (changes requested):
Thanks @veena14cs. Took a pass on the latest changes--PTAL.

veena14cs committed:
Fixed changes

BenHenning commented on app/src/main/res/values/styles.xml:
```diff
@@ -348,7 +352,7 @@
   </style>
 
-  <style name="CaptionViewCenter" parent="Caption">
+  <style name="CaptionViewCenterVertical" parent="Caption">
```
No views were updated when this was changed--is this actually being used anywhere?

veena14cs replied:
I have updated changes some Textviews were missed.

BenHenning commented on app/src/main/res/layout-land/profile_chooser_profile_view.xml:
```diff
@@ -61,26 +61,20 @@
 
         <TextView
           android:id="@+id/profile_is_admin_text"
+          style="@style/SubtitleLight1"
```
Should this be the centered version since the gravity was center before?

veena14cs replied:
Fixed.

BenHenning commented on app/src/main/res/layout/profile_chooser_add_view.xml:
```diff
@@ -59,6 +60,7 @@
 
         <TextView
           android:id="@+id/add_profile_description_text"
+          style="@style/TextViewStart"
```
Should this be centered per the layout_gravity? We did this for an earlier layout in the PR.

veena14cs replied:
`layout_gravity` seems unused here as gravity is controlled by parent layout, so removed it. The description is start aligned as per the UI.

BenHenning reviewed (commented):
Thanks @veena14cs. Just a few comments left. Some of them might be correct as-is per your earlier comments, but I just wanted to double check.

veena14cs committed:
Merge remote-tracking branch 'upstream/develop' into develop

veena14cs committed:
Merge branch 'develop' into section-topic-style

veena14cs committed:
fixed changes

veena14cs commented:
> Thanks @veena14cs. Just a few comments left. Some of them might be correct as-is per your earlier comments, but I just wanted to double check.

@BenHenning I have made changes and addressed your comments PTAL.

rt4914 commented:
@BenHenning and @veena14cs I have answered/resolved to all the comment threads that were started by me. Thanks.

rt4914 reviewed (approved):
LGTM, thanks.

BenHenning reviewed (approved):
Thanks @veena14cs! I have nothing else to add. I think this is a really nice change toward centrally managing RTL properties, and sets us up for more cleanly organizing centralized styles & enforcing their presence in future work.

BenHenning commented:
> @BenHenning and @veena14cs I have answered/resolved to all the comment threads that were started by me. Thanks.

Thanks @rt4914!

BenHenning commented:
Merging since there are no more open conversation threads & everything is approved.
